### PR TITLE
chore: update copyright to 2024

### PR DESCRIPTION
**Pull request by tmspzz**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Copyright notice is outdated

### Solutions

Update copyright to 2024

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2466](https://10.10.124.134/job/AR-build-pipeline/2466/) **failed**.

**Comment by AndroidBob** [comment]
Build [2467](https://10.10.124.134/job/AR-build-pipeline/2467/) **failed**.

File: build-logic/plugins/src/main/kotlin/AndroidCoordinates.kt [hunk 1]
```diff
@@ -1,8 +1,25 @@
+/*
```
Comment (yamilmedina) [inline build-logic/plugins/src/main/kotlin/AndroidCoordinates.kt:1]: duplicated copyright in this file

File: build-logic/plugins/src/main/kotlin/AndroidApplicationConventionPlugin.kt [hunk 2]
```diff
@@ -1,3 +1,20 @@
+/*
```
Comment (yamilmedina) [inline build-logic/plugins/src/main/kotlin/AndroidApplicationConventionPlugin.kt:1]: duplicated copyright in this file

File: build-logic/plugins/src/main/kotlin/AndroidLibraryConventionPlugin.kt [hunk 3]
```diff
@@ -1,3 +1,20 @@
+/*
```
Comment (yamilmedina) [inline build-logic/plugins/src/main/kotlin/AndroidLibraryConventionPlugin.kt:1]: duplicated copyright in this file

File: build-logic/plugins/src/main/kotlin/HiltConventionPlugin.kt [hunk 4]
```diff
@@ -1,10 +1,27 @@
+/*
```
Comment (yamilmedina) [inline build-logic/plugins/src/main/kotlin/HiltConventionPlugin.kt:1]: duplicated copyright in this file

File: build-logic/plugins/src/main/kotlin/com/wire/android/gradle/version/Versionizer.kt [hunk 5]
```diff
@@ -1,7 +1,24 @@
+/*
```
Comment (yamilmedina) [inline build-logic/plugins/src/main/kotlin/com/wire/android/gradle/version/Versionizer.kt:1]: duplicated copyright in this file

**Review by yamilmedina — APPROVED** [review]

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [2468](https://10.10.124.134/job/AR-build-pipeline/2468/) **failed**.

**Comment by AndroidBob** [comment]
Build [2470](https://10.10.124.134/job/AR-build-pipeline/2470/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8550914)(8550914)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8550914)(8550914)-staging-compat.apk)

**Pull request merged**